Error in handling `)"` in string literals when generating code in provenance phase

Here is a mini example `void.dl`:
```souffle
.type Function
.type Object
.decl invoke(?func: Function, ?obj: Object)
.input invoke(filename="invoke.facts")
.decl voidinvoke(?obj: Object)
.output voidinvoke
voidinvoke(obj) :- invoke("void()", obj).
```
It's okay when using `souffle -c void.dl` options to compile and run this program.
But when I trying to use the provenance, using `souffle -c void.dl -t1`. Error happens:
```shell
void.cpp:48:26: error: stray ‘@’ in program
    invoke("void()",obj,_,@level_num_0).)",
                          ^
void.cpp:48:41: warning: missing terminating " character
    invoke("void()",obj,_,@level_num_0).)",
                                         ^
void.cpp:47:2: error: missing terminating " character
  R"(voidinvoke(obj,1,(@level_num_0+1)) :- 
  ^
void.cpp: In constructor ‘souffle::Sf_void::Sf_void()’:
void.cpp:48:20: error: ‘obj’ was not declared in this scope
    invoke("void()",obj,_,@level_num_0).)",
                    ^
void.cpp:48:24: error: ‘_’ was not declared in this scope
    invoke("void()",obj,_,@level_num_0).)",
                        ^
void.cpp:48:27: error: ‘level_num_0’ was not declared in this scope
    invoke("void()",obj,_,@level_num_0).)",
                           ^
void.cpp:48:38: error: expected ‘}’ before ‘)’ token
    invoke("void()",obj,_,@level_num_0).)",
                                      ^
void.cpp:48:38: error: could not convert ‘{"void()", "invoke", "voidinvoke(obj,1,(@level_num_0+1)) :- \012   invoke(\"void(", <expression error>, <expression error>, <expression error>}’ from ‘<brace-enclosed initializer list>’ to ‘const std::vector<std::__cxx11::basic_string<char> >’
void.cpp:48:38: error: expected ‘,’ or ‘;’ before ‘)’ token
```

And in the line 43~49 of generated c++ file:
```cpp
// -- initialize symbol table --
static const std::vector<std::string> symbols = {
	R"(void())",
	R"(invoke)",
	R"(voidinvoke(obj,1,(@level_num_0+1)) :- 
   invoke("void()",obj,_,@level_num_0).)",
};
```
the `"void()"` is mistaken by cpp compiler as the closing of string literal. So I think the code-generator in provenance part needs correction to handle this.

--- a/src/Synthesiser.cpp
+++ b/src/Synthesiser.cpp
@@ -232,9 +232,9 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             out << "try {";
             out << "std::map<std::string, std::string> directiveMap(";
             out << load.getIODirectives() << ");\n";
-            out << R"(if (!inputDirectory.empty() && directiveMap["IO"] == "file" && )";
+            out << R"_(if (!inputDirectory.empty() && directiveMap["IO"] == "file" && )_";
             out << "directiveMap[\"filename\"].front() != '/') {";
-            out << R"(directiveMap["filename"] = inputDirectory + "/" + directiveMap["filename"];)";
+            out << R"_(directiveMap["filename"] = inputDirectory + "/" + directiveMap["filename"];)_";
             out << "}\n";
             out << "IODirectives ioDirectives(directiveMap);\n";
             out << "IOSystem::getInstance().getReader(";
@@ -254,9 +254,9 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             for (IODirectives ioDirectives : store.getIODirectives()) {
                 out << "try {";
                 out << "std::map<std::string, std::string> directiveMap(" << ioDirectives << ");\n";
-                out << R"(if (!outputDirectory.empty() && directiveMap["IO"] == "file" && )";
+                out << R"_(if (!outputDirectory.empty() && directiveMap["IO"] == "file" && )_";
                 out << "directiveMap[\"filename\"].front() != '/') {";
-                out << R"(directiveMap["filename"] = outputDirectory + "/" + directiveMap["filename"];)";
+                out << R"_(directiveMap["filename"] = outputDirectory + "/" + directiveMap["filename"];)_";
                 out << "}\n";
                 out << "IODirectives ioDirectives(directiveMap);\n";
                 out << "IOSystem::getInstance().getWriter(";
@@ -458,7 +458,7 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             const std::string ext = fileExtension(Global::config().get("profile"));
 
             // create local timer
-            out << "\tLogger logger(R\"(" << timer.getMessage() << ")\");\n";
+            out << "\tLogger logger(R\"_(" << timer.getMessage() << ")\"sf);\n";
 
             // insert statement to be measured
             visit(timer.getStatement(), out);
@@ -1295,7 +1295,7 @@ void Synthesiser::generateCode(const RamTranslationUnit& unit, std::ostream& os,
         os << "// -- initialize symbol table --\n";
         os << "static const std::vector<std::string> symbols = {\n";
         for (size_t i = 0; i < symTable.size(); i++) {
-            os << "\tR\"(" << symTable.resolve(i) << ")\",\n";
+            os << "\tR\"_(" << symTable.resolve(i) << ")_\",\n";
         }
         os << "};\n";
         os << "symTable.insert(symbols);\n";
@@ -1370,9 +1370,9 @@ void Synthesiser::generateCode(const RamTranslationUnit& unit, std::ostream& os,
             for (IODirectives ioDirectives : store->getIODirectives()) {
                 os << "try {";
                 os << "std::map<std::string, std::string> directiveMap(" << ioDirectives << ");\n";
-                os << R"(if (!outputDirectory.empty() && directiveMap["IO"] == "file" && )";
+                os << R"_(if (!outputDirectory.empty() && directiveMap["IO"] == "file" && )_";
                 os << "directiveMap[\"filename\"].front() != '/') {";
-                os << R"(directiveMap["filename"] = outputDirectory + "/" + directiveMap["filename"];)";
+                os << R"_(directiveMap["filename"] = outputDirectory + "/" + directiveMap["filename"];)_";
                 os << "}\n";
                 os << "IODirectives ioDirectives(directiveMap);\n";
                 os << "IOSystem::getInstance().getWriter(";
@@ -1401,9 +1401,9 @@ void Synthesiser::generateCode(const RamTranslationUnit& unit, std::ostream& os,
         os << "try {";
         os << "std::map<std::string, std::string> directiveMap(";
         os << load.getIODirectives() << ");\n";
-        os << R"(if (!inputDirectory.empty() && directiveMap["IO"] == "file" && )";
+        os << R"_(if (!inputDirectory.empty() && directiveMap["IO"] == "file" && )_";
         os << "directiveMap[\"filename\"].front() != '/') {";
-        os << R"(directiveMap["filename"] = inputDirectory + "/" + directiveMap["filename"];)";
+        os << R"_(directiveMap["filename"] = inputDirectory + "/" + directiveMap["filename"];)_";
         os << "}\n";
         os << "IODirectives ioDirectives(directiveMap);\n";
         os << "IOSystem::getInstance().getReader(";

